.github/workflows/ci-build.yaml

```diff
@@ -336,6 +336,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
+      - uses: cachix/install-nix-action@v22
```

Comment:
similarly as the above comments, this should be moved into `lint`
```suggestion
```

Reply:
The patch script also needs to be added to `changed-files`

Reply:
This is to determine if the lint job should run right? 
any file added there will trigger the lint job to run? 
I think in this case, the go.mod and go.sum files should also be added.

Reply:
> This is to determine if the lint job should run right?
> any file added there will trigger the lint job to run?

Correct, yes.

> I think in this case, the go.mod and go.sum files should also be added.

They're already there, [under `tests`](https://github.com/argoproj/argo-workflows/blob/13444e663387e3b5b331c278cd9e79fc88968d7e/.github/workflows/ci-build.yaml#L54), which is [inherited by `lint`](https://github.com/argoproj/argo-workflows/blob/13444e663387e3b5b331c278cd9e79fc88968d7e/.github/workflows/ci-build.yaml#L87)

Reply:
Ah okay thanks